src/storage/FileResourceMapper.ts

```diff
@@ -0,0 +1,48 @@
+import { types } from 'mime-types';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { FileResourceStore } from './FileResourceStore';
+import { ResourceMapper } from './ResourceMapper';
+
+export class FileResourceMapper implements ResourceMapper {
+  private readonly fileStore: FileResourceStore;
+  private readonly types: Record<string, any>;
+
+  public constructor(fileStore: FileResourceStore, overrideTypes = { acl: 'text/turtle', metadata: 'text/turtle' }) {
+    this.fileStore = fileStore;
+    this.types = { ...types, ...overrideTypes };
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   */
+  public mapUrlToFilePath(identifier: ResourceIdentifier): string {
+    if (!identifier.path.startsWith(this.fileStore.baseRequestURI)) {
+      throw new NotFoundHttpError();
+    }
+    return identifier.path.slice(this.fileStore.baseRequestURI.length);
+  }
+
+  /**
+   * Strips the rootFilepath path from the filepath and adds the baseRequestURI in front of it.
+   * @param path - The filepath.
+   *
+   * @throws {@Link Error}
+   * If the filepath does not match the rootFilepath path of the store.
+   */
+  public mapFilePathToUrl(path: string): string {
+    if (!path.startsWith(this.fileStore.rootFilepath)) {
+      throw new Error(`File ${path} is not part of the file storage at ${this.fileStore.rootFilepath}.`);
+    }
+    return this.fileStore.baseRequestURI + path.slice(this.fileStore.rootFilepath.length);
+  }
+
+  public getContentTypeFromExtension(path: string): string {
+    const extension = /\.([^./]+)$/u.exec(path);
+    return (extension && this.types[extension[1].toLowerCase()]) || false;
```

Comment:
The `false` here is weird.

Reply:
Especially since the return type is `string` so I'm not even sure how this passes 😄

Reply:
oops, better if it returns an empty string in stead of false

Reply:
or undefined rather, and change return type

Reply:
Or throw an error even? Since I don't think we ever want to have no content-type (incoming data with no content-type should have been rejected already).

Reply:
Or default to `application/octet-stream`.

Reply:
Yea that also solves it

Reply:
defaulting to applicatoin/octet-stream would makes a unit test fail: 'A FileResourceStore › returns no contentType when unknown for representation.'  Which is why i didnt use a default contenttype, but I can just change the test then?

Reply:
Let's change indeed (and add `APPLICATION_OCTET_STREAM` to `ContentTypes.ts`)

Reply:
also just a thought: contentType is an optional field in RepresentationMetadata, but it sounds like there should always be a contentType?

Reply:
Should; perhaps just might not always be known, @joachimvh?

Reply:
Input without a content-type already gets rejected [here](https://github.com/solid/community-server/blob/master/src/ldp/http/RawBodyParser.ts#L29) so if we don't have a content-type something went wrong somewhere. How metadata is represented is going to change drastically soon so it doesn't matter that much that the field is still optional now.